kylin-spark-project/kylin-spark-engine/src/main/java/org/apache/kylin/engine/spark/job/NSparkExecutable.java

```diff
@@ -382,6 +387,8 @@ protected String generateSparkCmd(KylinConfig config, String hadoopConf, String
         if (StringUtils.isNotBlank(sparkUploadFiles)) {
             sb.append("--files ").append(sparkUploadFiles).append(" ");
         }
+        sb.append("--principal ").append(config.getKerberosPrincipal()).append(" ");
+        sb.append("--keytab ").append(config.getKerberosKeytabPath()).append(" ");
```

Comment:
why it needs these two lines? please check the code in the method 'replaceSparkNodeJavaOpsConfIfNeeded', it will set the parameters about kerberos according to 'config.isKerberosEnabled()'

Reply:
My mistake, I'll add Kerberos usage check right here, but not in java options.